### PR TITLE
OHRM5X-1650: Develop dashboard - card component

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/client/src/orangehrmDashboardPlugin/components/ActionButton.vue [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export default {
+  name: 'ActionButton',
+  components: {
+    'oxd-icon': Icon,
+  },
+  emits: ['show-actions'],
+};
+</script>
+<style lang=""></style>
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/ActionButton.vue:37]:
```suggestion
```

File: src/client/src/orangehrmDashboardPlugin/components/ActionButton.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div>
+    <oxd-icon name="gear-fill" @click="$emit('show-actions', true)"></oxd-icon>
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/ActionButton.vue:23]: What if we remove outer div and pass the icon name as prop?

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+<template>
+  <oxd-sheet :gutters="false" class="orangehrm-dashboard-widget">
+    <div class="orangehrm-dashboard-widget-header">
+      <div class="orangehrm-dashboard-widget-name">
+        <oxd-icon
+          class="orangehrm-dashboard-widget-icon"
+          :name="iconName"
+        ></oxd-icon>
+        <slot name="widget-settings"></slot>
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:29]: this slot seems not used

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        <oxd-text v-show="widgetName" type="card-title">
+          {{ widgetName }}
+        </oxd-text>
+      </div>
+      <div>
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:34]: remove the empty div

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+          :src="defaultPic"
+          alt="No Content"
+          class="orangehrm-dashboard-widget-img"
+        />
+        <oxd-text tag="p">
+          {{ emptyContentText }}
+        </oxd-text>
+      </div>
+    </div>
+    <oxd-loading-spinner v-if="isLoading" class="orangehrm-container-loader" />
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:55]: If it its loading, body slot should be hidden.

```
<oxd-loading-spinner v-if="isLoading" class="orangehrm-container-loader" />
<div v-else class="orangehrm-dashboard-widget-body">
  <slot name="body"></slot>
 ...
 ...
</div>
```

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  </oxd-sheet>
+</template>
+<script>
+import Sheet from '@ohrm/oxd/core/components/Sheet/Sheet';
+import Icon from '@ohrm/oxd/core/components/Icon/Icon';
+
+export default {
+  name: 'WidgetCard',
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:63]: Rename this component as BaseWidget (BaseWidget.vue)

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    'oxd-icon': Icon,
+  },
+  props: {
+    iconName: {
+      type: String,
+      required: true,
+    },
+    widgetName: {
+      type: String,
+      default: '',
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:75]: Required.

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+  components: {
+    'oxd-sheet': Sheet,
+    'oxd-icon': Icon,
+  },
+  props: {
+    iconName: {
+      type: String,
+      required: true,
+    },
+    widgetName: {
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:73]:
```suggestion
    title: {
```

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+import Icon from '@ohrm/oxd/core/components/Icon/Icon';
+
+export default {
+  name: 'WidgetCard',
+  components: {
+    'oxd-sheet': Sheet,
+    'oxd-icon': Icon,
+  },
+  props: {
+    iconName: {
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:69]:
```suggestion
    icon: {
```

File: src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+      return !!this.$slots.body;
+    },
+    hasActionSlot() {
+      return !!this.$slots.action;
+    },
+  },
+};
+</script>
+<style lang="scss" scoped>
+@import '@ohrm/oxd/styles/_mixins.scss';
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/CardComponent.vue:104]: let's move this as separate stylesheet.

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmDashboardPlugin/components/BaseWidget.vue [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    'oxd-loading-spinner': Spinner,
+  },
+  props: {
+    icon: {
+      type: String,
+      required: true,
+    },
+    title: {
+      type: String,
+      default: '',
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/BaseWidget.vue:77]: title should be required

File: src/client/src/orangehrmDashboardPlugin/components/BaseWidget.vue [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    },
+    emptyContentText: {
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/BaseWidget.vue:79]: Can we have generic text for this? this prop is not required

**Review by Super-Chama — CHANGES_REQUESTED** [review]

**Review by Super-Chama — APPROVED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**